xunit.runner.visualstudio.desktop/VsTestRunner.cs

```diff
@@ -85,9 +85,22 @@ void ITestDiscoverer.DiscoverTests(IEnumerable<string> sources, IDiscoveryContex
             RunSettingsHelper.ReadRunSettings(discoveryContext?.RunSettings?.SettingsXml);
 #endif
 
+            var testPlatformContext = new TestPlatformContext
```

Comment:
In all 3 operations from VSTest: Discover, Run All, Run Specific tests, this context is defined. The optimizations are as follows:

| | Discovery | Run All | Run Specific |
|-|--------------|----------|-----------------|
| RequiresSourceInformation | IDE - Yes, CLI - No | IDE - Yes, CLI - No | IDE - Yes, CLI - No |
| RequireXunitTestProperty | IDE - Yes, CLI - Yes* | IDE - Yes, CLI - No | IDE - Yes, CLI - No |

CLI can be optimized since it is fire and forget. Except for Discovery* since `Run Specific Test` in CLI also calls Discovery first and then followed by an execution.